modules/new_user_cog.py

```diff
@@ -0,0 +1,51 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Thu Jan  7 09:53:09 2021
+
+@author: pinny
+"""
+from discord import utils
+from discord.ext import commands
+from modules.new_user.new_user import *
+
+class NewUser(commands.Cog):
+    def __init__ (self, bot):
+        self.bot = bot
+
+
+    @commands.command(name='join')
+    async def join(self, ctx, first_name : str, last_name : str, machon : str, year : str):
+        '''Join command to get new users information and place them in the right roles'''        
+        
+        #user who wrote the command
+        member = ctx.author
+        
+        if is_new(member) != True:
+            return
+    
+        await add_role(ctx, machon, year)
+        await remove_unassigned(member)
+        await change_nick(member, first_name, last_name)
+            
+    
+    @commands.Cog.listener()
+    async def on_command_error(self, ctx, error):
+        '''catch missing required argument error'''
+        
+        if isinstance(error, commands.MissingRequiredArgument):
+            await ctx.send("One or multiple required arguments are missing, please try again.")
+    
+    
+    @commands.Cog.listener()
+    async def on_member_join(self, member):
+        '''Welcome members who join.'''
+        print(f"{member.name} joined the guild.")
+        
+        #Sets the channel to the welcome channel and sends a message to it
+        channel = utils.get(member.guild.channels, name = ":wave:welcome")#I'm not positive this is what the channel is called
+        await channel.send(f"Welcome, {member.name}, please use the join command to continue. ~join <firstname> <lastname> <machon Lev/Tal> <year of graduation>")
```

Comment:
People might not know their year of graduation offhand. May want to give a key as to which year number (1st, 2nd, 3rd) corresponds to which year (or just allow them to type their year number and have the bot figure it out.

Reply:
I was thinking the interface could be something like this:

Welcome to the server <user>!
Please type the following command so we know who you are:

~join **first-name**, **last-name**, **campus**, **year**

Where:
- **first-name** is your first name,
- **last-name** is your last name,
- **campus** is one of "Lev" or "Tal" (no quotes, case insensitive),
- **year** is one of 1, 2, 3, or 4